packages/chart/src/ChartThemeProvider.tsx

```diff
@@ -0,0 +1,38 @@
+import { createContext, ReactNode, useEffect, useState } from 'react';
+import { useTheme } from '@deephaven/components';
+import initChartTheme, { ChartTheme } from './ChartTheme';
+
+export type ChartThemeContextValue = ChartTheme;
+
+export const ChartThemeContext = createContext<ChartThemeContextValue | null>(
+  null
+);
+
+export interface ChartThemeProviderProps {
+  children: ReactNode;
+}
+
+/*
+ * Provides a chart theme based on the active themes from the ThemeProvider.
+ */
+export function ChartThemeProvider({
+  children,
+}: ChartThemeProviderProps): JSX.Element {
+  const { activeThemes } = useTheme();
+
+  const [chartTheme, setChartTheme] = useState<ChartTheme | null>(null);
+
+  // Running in an effect to ensure parent ThemeProvider has had a chance to add
+  // the <style> tags to the DOM that provide theme variables
```

Comment:
Not sure this comment is really necessary - if the `activeThemes` ever change, we need to re-initialize the chart theme, whether the theme provider has had a chance to add the style tags or not.
That being said, perhaps the `ThemeProvider` should gate until it has loaded?

Reply:
I actually tried that before, but it breaks Enterprise due to the way the component tree is structured.

Provider
  ThemeProvider
    MonacoBootstrap (depends on theming)
      ..
      App
        AppMainContainer (depends on theming)
        Login (can't block this from rendering)

Reply:
It's possible the `ChartThemeProvider` could be moved further down in the tree, but it would mean we can't bundle theming providers together such as the `ThemeBootstrap` does

Reply:
Also to clarify, the comment is distinguishing that we have to load in useEffect to ensure any changes produced by setting the theme have completed. This would also apply if the theme is changed. This is in contrast to useMemo which will attempt to process the newly set theme before the style tags have actually been added and made the variables available. I can improve the comment if it helps, or let me know if there is something else I am missing.